File: app/src/main/java/ds/vkplus/model/PhotoData.java
package ds.vkplus.model;

public class PhotoData {

	public static final int TYPE_PHOTO = 0;
	public static final int TYPE_VIDEO = 1;
	public static final int TYPE_LINK = 2;

	public final long id;
	public int width;
	public int height;
	public String url;
	public float ratio;
	public int type;
	public boolean breakAfter;
	public boolean floating;
	public boolean paddingBottom;
	public String extra;


	public PhotoData(final String url, final int width, final int height, final int t, final long id) {
		this.width = width != 0 ? width : 1600;
		this.height = height != 0 ? height : 1200;
		this.url = url;
		ratio = (float) width / height;
		type = t;
		this.id = id;
	}


	public PhotoData(final String url, final String urlBig, final long id) {
		this.url = url;
		extra = urlBig;
		this.id = id;
	}


	public void setViewSize(final float w, final float h, final boolean breakAfter, final boolean floating) {
		width = (int) w;
		height = (int) h;
		this.breakAfter = breakAfter;
		this.floating = floating;


	}
}

File: app/src/main/java/ds/vkplus/model/PhotoData.kt
package ds.vkplus.model

class PhotoData {
	
	val id: Long
	var width = 0
	var height = 0
	var url: String
	var ratio = 0F
	var type = 0
	var breakAfter = false
	var floating = false
	//var paddingBottom = false
	var extra: String?=null
	
	constructor(url: String, width: Int, height: Int, t: Int, id: Long) {
		this.width = if (width != 0) width else 1600
		this.height = if (height != 0) height else 1200
		this.url = url
		ratio = width.toFloat() / height
		type = t
		this.id = id
	}
	
	
	constructor(url: String, urlBig: String, id: Long) {
		this.url = url
		extra = urlBig
		this.id = id
	}
	
	
	fun setViewSize(w: Float, h: Float, breakAfter: Boolean, floating: Boolean) {
		width = w.toInt()
		height = h.toInt()
		this.breakAfter = breakAfter
		this.floating = floating
		
		
	}
	
	companion object {
		
		val TYPE_PHOTO = 0
		val TYPE_VIDEO = 1
		val TYPE_LINK = 2
	}
}
